src/prefect/tasks/kubernetes/job.py

```diff
@@ -695,34 +698,68 @@ def run(
                 "The job name must be defined in the body under the metadata key."
             )
 
-        api_client = cast(
+        api_client_job = cast(
             client.BatchV1Api, get_kubernetes_client("job", kubernetes_api_key_secret)
         )
 
-        api_client.create_namespaced_job(namespace=namespace, body=body, **kube_kwargs)
+        api_client_pod = cast(
+            client.CoreV1Api, get_kubernetes_client("pod", kubernetes_api_key_secret)
+        )
+
+        api_client_job.create_namespaced_job(
+            namespace=namespace, body=body, **kube_kwargs
+        )
         self.logger.info(f"Job {job_name} has been created.")
 
-        completed = False
-        while not completed:
-            res = api_client.read_namespaced_job_status(
-                name=job_name, namespace=namespace
```

Comment:
Since:
- not all users will want to stream logs
- relying on a log stream for job completion isn't sufficient

I think we'll still want to check for job completion via the job status. The polling loop is one way, starting a watch stream on the job status would be another (I could go either way here).